Source/WebGPU/WGSL/Types.cpp

```diff
@@ -40,7 +40,7 @@ using namespace Types;
 
 void Type::dump(PrintStream& out) const
 {
-    WTF::switchOn(*this,
+    WTF::switchOn(static_cast<const TypeVariant>(*this),
```

Comment:
is there any way we can avoid these static_casts by changing the implementation or definition of `switchOn`?